components/nimbus/src/stateful/nimbus_client.rs

```diff
@@ -803,6 +803,78 @@ impl NimbusClient {
         Ok(Vec::new())
     }
 
+    pub fn register_previous_gecko_pref_states(
+        &self,
+        gecko_pref_states: &[GeckoPrefState],
+    ) -> Result<()> {
+        let previous_states = super::gecko_prefs::build_previous_states(gecko_pref_states);
+
+        let db = self.db()?;
+        let mut writer = db.write()?;
+
+        for (experiment_slug, previous_state) in previous_states {
+            Self::add_previous_state_for_experiment(
+                db,
+                &mut writer,
+                &experiment_slug,
+                PreviousState::GeckoPref(previous_state),
+            )?;
+        }
+
+        let mut state = self.mutable_state.lock().unwrap();
+        self.end_initialize(db, writer, &mut state)?;
+        Ok(())
+    }
+
+    pub(crate) fn add_previous_state_for_experiment(
+        db: &Database,
+        writer: &mut Writer,
+        experiment_slug: &str,
+        previous_state: PreviousState,
+    ) -> Result<()> {
+        let enrollments = db.get_store(StoreId::Enrollments);
+
+        if let Ok(Some(existing_enrollment)) =
+            enrollments.get::<ExperimentEnrollment, Writer>(writer, experiment_slug)
+        {
+            // Previous state is only valid on Enrolled experiments
+            let updated_state = existing_enrollment.on_add_state(previous_state);
+            enrollments.put(writer, experiment_slug, &updated_state)?;
+        }
+        Ok(())
+    }
+
+    pub fn register_previous_state(
```

Comment:
This seems unused outside tests and its not available in the UDL. We can get rid of this.